### PR TITLE
[Intel MKL] Updating MatMul kernels with MKLDNN 1.x API changes

### PR DESCRIPTION
**Pull request by nhasabni**
This PR updates QMatMul and FusedMatMul MKL CPU kernels with MKLDNN 1.0 API.
It also updates MatMul and BatchMatMul BFloat16 kernels for MKL CPU with MKLDNN 1.2 API.
Some of the changes are suggested by clang formet check tool 8.0.1 version.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_qmatmul_op.cc [hunk 1]
```diff
@@ -522,12 +554,17 @@ class MklDnnQuantizedMatMulOp : public MklDnnMatMulOpBase<Toutput> {
...
                                 DT_INT32, weight_mkl_format, &weight_oi_md,
                                 &weight_md_tensor_ptr));
+#ifdef ENABLE_MKLDNN_V1
+    // Using the logic from filter caching in mkl_conv_ops.cc
+    weight_md_tensor_ptr->scalar<int32>()() =
+        static_cast<int32>(weight_mkl_shape.GetTfDataFormat());
+#else
     weight_md_tensor_ptr->scalar<int32>()() =
         matmul_fwd_pd.get()->weights_primitive_desc().desc().data.format;
+#endif  // ENABLE_MKLDNN_V1
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_qmatmul_op.cc]: We can use
```c++
weight_md_tensor_ptr->scalar<int32>()() = 
    static_cast<int32>(GET_TF_DATA_FORMAT(weight_mkl_shape, matmul_fwd_pd.get()->weights_primitive_desc().desc());
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

**Comment by nhasabni** [comment]
hi @penpornk, thanks for review. I've addressed your comments. Pls take a look.

File: tensorflow/core/kernels/mkl_matmul_ops_common.h [hunk 2]
```diff
@@ -103,8 +113,13 @@ class MklDnnMatMulFwdPrimitive : public MklPrimitive {
     context_.dst_mem->set_data_handle(DummyData);
   }
 
+#ifndef ENABLE_MKLDNN_V1
+  // In MKL-DNN v1.x, memory format tags only provide a partial description
+  // of the memory layout. Hence, these functions are disabled for v1.x.
   memory::format GetSrcMemoryFormat() const { return context_.src_fmt; }
   memory::format GetweightMemoryFormat() const { return context_.weight_fmt; }
+#endif  // ENABLE_MKLDNN_V1
```
[the remark this answers is not in the document]
Comment (penpornk) [reply tensorflow/core/kernels/mkl_matmul_ops_common.h]: I think you missed this one. :)

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the quick fixes! I have two more comments.

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc [hunk 3]
```diff
@@ -174,122 +174,105 @@ class BatchMatMulMkl : public OpKernel {
...
+      std::vector<double> alpha_Array(group_size[0], 1.0);
+      std::vector<double> beta_Array(group_size[0], 0.0);
+      cblas_dgemm_batch(
+          Layout, &TransA_Array[0], &TransB_Array[0], &M_Array[0], &N_Array[0],
+          &K_Array[0], &alpha_Array[0],
+          reinterpret_cast<const double**>(A_Array), &lda_Array[0],
+          reinterpret_cast<const double**>(B_Array), &ldb_Array[0],
+          &beta_Array[0], reinterpret_cast<double**>(C_Array), &ldc_Array[0],
+          group_count, &group_size[0]);
+    }
```
[the remark this answers is not in the document]
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_batch_matmul_op.cc]: Yes, I thought about using T for all the MklBlasBatchGemm functions. The problems though are 1) each type is using different gemm call and input transpose params (that means using runtime check which would introduce unnecessary slowdown) , and 2) implementation for complex types is slightly different than float and double -- input tensors to actual gemm call are void** type for complex while they are T** for float and double.

File: tensorflow/core/kernels/mkl_matmul_ops_common.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (nhasabni) [reply tensorflow/core/kernels/mkl_matmul_ops_common.h]: :) Done.

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (penpornk) [reply tensorflow/core/kernels/mkl_batch_matmul_op.cc]: I see. I forgot about `sgemm` vs `dgemm` calls here. Thank you for the explanation!

**Review by penpornk — DISMISSED** [review]
Thank you for the additional fixes as well! :)

**Comment by nhasabni** [comment]
@penpornk Thanks a lot for quick review and approval!

**Comment by gbaned** [comment]
@nhasabni Can you please resolve conflicts? Thanks!

**Comment by nhasabni** [comment]
@penpornk I've resolved merge conflict. Pls take a look.

**Review by penpornk — APPROVED** [review]
Thank you very much!

**Pull request merged**